### PR TITLE
Alpha support to Elliptic Curve keys to JsonWebKey classes

### PR DESCRIPTION
**Pull request by msfcolombo**
## Alpha support to Elliptic Curve keys to JsonWebKey classes ###

This PR adds alpha support for Elliptic Curve keys the Microsoft.Azure.KeyVault.WebKey namespace. This is intended to allow customers and partners accessing the Elliptic Curve features of Azure Key Vault in preview mode.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.WebKey.Tests/WebKeyEcValidationTest.cs
Comment (shahabhijeet) [inline src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.WebKey.Tests/WebKeyEcValidationTest.cs]: @msfcolombo please use #FullNetFx moniker for Full Desktop version of .NET

File: src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.WebKey/Microsoft.Azure.KeyVault.WebKey.csproj
Comment (shahabhijeet) [inline src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.WebKey/Microsoft.Azure.KeyVault.WebKey.csproj]: @msfcolombo any reason to start taking dependency on common clientRuntime binaries. Only management plane has taken dependency on them?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@msfcolombo please have a signoff from include Rich Randal or Dragos

File: src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.WebKey/Microsoft.Azure.KeyVault.WebKey.csproj
Comment (msfcolombo) [inline src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.WebKey/Microsoft.Azure.KeyVault.WebKey.csproj]: We intend to provide WebKey to more environments. It's a low-level component. The use of AzSdk.reference.props is to insure compatibility and csproj code duplication.

**Review by herveyw-msft — APPROVED** [review]
Approved.

**Review by dragav — APPROVED** [review]
Approved.

**Comment by msfcolombo** [comment]
@shahabhijeet I did the change you requested, and commented over usage of AzSdk.reference.props. I got sign-off from Hervey and Dragos. Is there something else before merging?

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by msfcolombo** [comment]
@shahabhijeet All requested changes were performed. Thanks for the push into keeping a low number of dependencies, and let us know if there is anything preventing merge.

**Comment by msfcolombo** [comment]
Changes here are now included on #3577.

**Pull request closed without merging**
